### PR TITLE
cpu/gd32v: add periph_rtt support

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the `periph_rtt` support and is one of a bunch of PRs that complete the peripheral drivers for GD32VF103.

### Testing procedure

`tests/periph_rtt` should work and give the following results:
```
Help: Press s to start test, r to print it is ready
START
main(): This is RIOT! (Version: 2023.04-devel-199-g2d429-cpu/gd32v/periph_rtt)

RIOT RTT low-level driver test
RTT configuration:
RTT_MAX_VALUE: 0xffffffff
RTT_FREQUENCY: 32768

Testing the tick conversion
Trying to convert 1 to seconds and back
Trying to convert 256 to seconds and back
Trying to convert 65536 to seconds and back
Trying to convert 16777216 to seconds and back
Trying to convert 2147483648 to seconds and back
All ok

Initializing the RTT driver
This test will now display 'Hello' every 5 seconds

RTT now: 148
Setting initial alarm to now + 5 s (163988)
rtt_get_alarm() PASSED
Done setting up the RTT, wait for many Hellos
{ "threads": [{ "name": "idle", "stack_size": 256, "stack_used": 216 }]}
{ "threads": [{ "name": "main", "stack_size": 1280, "stack_used": 480 }]}
Hello
Hello
Hello
Hello
```

### CODE REVIEW

File: cpu/gd32v/periph/rtt.c [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+/* RTC->ALRMH and RTC->ALRML are writable only. Therefore the current alarm
+ * time must be stored separately in a variable for rtt_get_alarm. */
+static uint32_t _rtt_alarm = 0;
+
+uint32_t rtt_get_alarm(void)
+{
+    /* wait for synchronization */
+    while (!(RTT_DEV->CTL & RTC_CTL_RSYNF_Msk)) { }
+
+    return  _rtt_alarm;
```
Comment (benpicco) [inline cpu/gd32v/periph/rtt.c:142]:
```suggestion
    return _rtt_alarm;
```

**Review by benpicco — APPROVED** [review]
looks good to me!

File: cpu/gd32v/periph/rtt.c (hunk 1 above)
Comment (gschorcht) [reply cpu/gd32v/periph/rtt.c:142]: I squashed it directly.

**Comment by benpicco** [comment]
bors merge

**Pull request merged**